### PR TITLE
Add recording file sharing switch

### PR DESCRIPTION
**Pull request by zbettenbuk**

### CODE REVIEW

File: react/features/recording/components/Recording/AbstractStartRecordingDialog.js [hunk 1]
```diff
@@ -31,6 +31,12 @@ type Props = {
      */
     _fileRecordingsServiceEnabled: boolean,
 
+    /**
+     * Whether to show the possibility to share file recording with other people (e.g. meeting participants), based on
+     * the actual implementation on the backend.
+     */
+    _fileRecordingsServiceSharingEnabled: Boolean,
```
Comment (saghul) [inline react/features/recording/components/Recording/AbstractStartRecordingDialog.js]: `boolean` ?

File: react/features/recording/components/Recording/AbstractStartRecordingDialog.js [hunk 2]
```diff
@@ -69,6 +75,11 @@ type State = {
      */
     selectedRecordingService: ?string,
 
+    /**
+     * True of the user requested the service to share the recording with others.
```
Comment (saghul) [inline react/features/recording/components/Recording/AbstractStartRecordingDialog.js]: typo: if the user

File: react/features/recording/components/Recording/AbstractStartRecordingDialog.js [hunk 3]
```diff
@@ -266,12 +297,16 @@ class AbstractStartRecordingDialog extends Component<Props, State> {
  * @returns {{
  *     _appKey: string,
  *     _conference: JitsiConference,
+ *     _fileRecordingsServiceEnabled: boolean,
+ *     _isDropboxEnabled: boolean,
+ *     _fileRecordingsServiceSharingEnabled: boolean,
```
Comment (saghul) [inline react/features/recording/components/Recording/AbstractStartRecordingDialog.js]: Maybe use `_isFileRecording...` for consistency with `_isDropboxEnabled` ? At any rate, alpha sort.

File: react/features/recording/components/Recording/native/StartRecordingDialog.js [hunk 4]
```diff
@@ -28,10 +28,15 @@ class StartRecordingDialog extends AbstractStartRecordingDialog {
...
             isValidating,
             selectedRecordingService,
+            sharingEnabled,
             spaceLeft,
             userName
         } = this.state;
-        const { _fileRecordingsServiceEnabled, _isDropboxEnabled } = this.props;
+        const {
+            _fileRecordingsServiceEnabled,
+            _fileRecordingsServiceSharingEnabled,
```
Comment (saghul) [inline react/features/recording/components/Recording/native/StartRecordingDialog.js]: same as above, let's find some more consistent naming if possible

File: react/features/recording/components/Recording/AbstractStartRecordingDialog.js (hunk 1 above)
Comment (zbettenbuk) [reply react/features/recording/components/Recording/AbstractStartRecordingDialog.js]: Autocomplete :(

Thanks!

File: react/features/recording/components/Recording/AbstractStartRecordingDialog.js (hunk 3 above)
Comment (zbettenbuk) [reply react/features/recording/components/Recording/AbstractStartRecordingDialog.js]: I got this name from Damyan, this is the actual config name that he'll use in config.js

Comment (saghul) [reply react/features/recording/components/Recording/AbstractStartRecordingDialog.js]: ok

File: react/features/recording/components/Recording/AbstractStartRecordingDialog.js [hunk 5]
```diff
@@ -280,6 +315,7 @@ export function mapStateToProps(state: Object) {
         _conference: state['features/base/conference'].conference,
         _fileRecordingsServiceEnabled: fileRecordingsServiceEnabled,
         _isDropboxEnabled: isDropboxEnabled(state),
+        _fileRecordingsServiceSharingEnabled: fileRecordingsServiceSharingEnabled,
```
Comment (saghul) [inline react/features/recording/components/Recording/AbstractStartRecordingDialog.js]: alpha sort

Comment (zbettenbuk) [reply react/features/recording/components/Recording/AbstractStartRecordingDialog.js]: Sh*t, I'm really off today...

**Pull request merged**